src/main/javascript/index.js

```diff
@@ -93,26 +92,51 @@ function addProperties(properties, requiredProperties, file) {
         fs.appendFileSync(file, '         */\n');
 
         if (_.some(requiredProperties, (el) => _.includes(name, el))) {
-            fs.appendFileSync(file, '        this.' + name + ' = ' + name + ';\n\n');
+            fs.appendFileSync(file, '        this.' + name + ' = builder.' + name + ';\n\n');
         } else {
             fs.appendFileSync(file, '        this.' + name + ' = undefined;\n\n');
         }
 
     });
 }
 
-function getParentClass(allOf) {
-    var parts = allOf['$ref'].split('/');
-    return parts[parts.length - 1];
+function addConstructor(definition, classname, file) {
+    fs.appendFileSync(file, '    /**\n');
+    fs.appendFileSync(file, '     * Constructs a ' + classname + ' based on its enclosed builder.\n\n');
+    fs.appendFileSync(file, '     * @constructor \n');
+    fs.appendFileSync(file, `     * @param {Builder} the ${classname} builder\n`);
+    fs.appendFileSync(file, '     */\n');
+    fs.appendFileSync(file, '    constructor(builder) {\n');
```

Comment:
Do we still need a constructor? Usually if there is a builder, one just writes for example `let product = new Builder().withParam1(param1).build();` ... there shouldn't be any constructor for the main class.

And the `build()` method would then instantiate the main class ... IMO your approach is not standard.

Reply:
For the builder I wrote the usage is `let cart = new Cart.Builder().withId(cifId).withEntries(cartEntries).withProductTotalPrice(productTotalPrice).withCurrency(currency).build();`
For me looks standard.
If we remove the constructor we will still have the default one (which is without params). With my proposal our model classes can be instantiated only with the Builder which is really powerfull AFAICS because we can easily add then validations in our build method (before returning the cart).

Reply:
yes that's ok, but for example the constructor `new Product(builder)` is IMO not standard. Usually there isn't any public constructor so one can only build the class with the builder, but my experience comes from Java, not so sure about Javascript ...

Reply:
but in java script everything is public. we can only achieve private if we use closure but then we would drop the class syntax.

Reply:
yes sure, I googled a bit and realised JS is quite permissive here ...